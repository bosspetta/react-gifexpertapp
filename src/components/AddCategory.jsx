import React, { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if (inputValue.trim().length <= 1) return
        // setCategories(prev => [inputValue, ...prev])
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                aria-label="Buscar GIF's por categorÃ­a"
                placeholder='Buca un GIF'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
