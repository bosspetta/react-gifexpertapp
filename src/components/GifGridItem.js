import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({id, title, url}) => {
  // console.log({id, title, url});
  return (
    <div className="card animate__animated animate__fadeIn">
      <img id={id} src={url} alt={title} />
      <p>
        <small>
          <strong>{title}</strong>
        </small>
      </p>
    </div>
  );
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
