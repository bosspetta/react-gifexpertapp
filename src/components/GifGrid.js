// import React, { useState, useEffect } from 'react';
import React from 'react';
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">
          <img
            src="https://www.ipgvip.com/wp-content/themes/ipg/assets/images/loading-gif.gif"
            alt="Icono de carga"
          />
          <br />
          Cargandoâ€¦
        </p>
      )}
      <li className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </li>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
