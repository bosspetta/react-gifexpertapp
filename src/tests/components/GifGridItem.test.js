import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Un título';
  const url   = 'http://localhost:8080';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe mostrar corectamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un párrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe de obtener el src y el alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener la clase animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
  
});
