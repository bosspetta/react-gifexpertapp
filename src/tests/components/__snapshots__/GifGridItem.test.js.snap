// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar corectamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="http://localhost:8080"
  />
  <p>
    <small>
      <strong>
        Un título
      </strong>
    </small>
  </p>
</div>
`;
